structural and content changes to the sql_database doc (#1623)
dlt-hub/dlt: docs/website/docs/dlt-ecosystem/verified-sources/sql_database/advanced.md

Incremental loading uses a cursor column (e.g., timestamp or auto-incrementing ID) to load only data newer than a specified initial value, enhancing efficiency by reducing processing time and resource use. Read [here](https://dlthub.com/docs/walkthroughs/sql-incremental-configuration) for more details on incremental loading with `dlt`.

#### How to configure
1. **Choose a Cursor Column**: Identify a column in your SQL table that can serve as a reliable indicator of new or updated rows. Common choices include timestamp columns or auto-incrementing IDs.
1. **Set an Initial Value**: Choose a starting value for the cursor to begin loading data. This could be a specific timestamp or ID from which you wish to start loading data.
1. **Deduplication**: When using incremental loading, the system automatically handles the deduplication of rows based on the primary key (if available) or row hash for tables without a primary key.
1. **Set end_value for backfill**: Set `end_value` if you want to backfill data from
certain range.
1. **Order returned rows**. Set `row_order` to `asc` or `desc` to order returned rows.

#### Examples

> 1. Incremental loading with the resource `sql_table`  

Reviewer: ```suggestion
**1. Incremental loading with the resource `sql_table`**  

```

    Consider a table "family" with a timestamp column `last_modified` that indicates when a row was last modified. To ensure that only rows modified after midnight (00:00:00) on January 1, 2024, are loaded, you would set `last_modified` timestamp as the cursor as follows:
   ```py
   from sql_database import sql_table
   from datetime import datetime

   # Example: Incrementally loading a table based on a timestamp column
   table = sql_table(
       table='family',
       incremental=dlt.sources.incremental(
           'last_modified',  # Cursor column name
           initial_value=pendulum.DateTime(2024, 1, 1, 0, 0, 0)  # Initial cursor value
       )
   )

   info = pipeline.extract(table, write_disposition="merge")
   print(info)
   ```
    Behind the scene, the loader generates a SQL query filtering rows with `last_modified` values greater than the incremental value. In the first run, this is the initial value (midnight (00:00:00) January 1, 2024). 
    In subsequent runs, it is the latest value of `last_modified` that `dlt` stores in [state](https://dlthub.com/docs/general-usage/state).

> 2. Incremental loading with the source `sql_database`  

Reviewer: ```suggestion
**2. Incremental loading with the source `sql_database`**  
 
```

    To achieve the same using the `sql_database` source, you would specify your cursor as follows:

    ```py
    source = sql_database().with_resources("family")
    #using the "last_modified" field as an incremental field using initial value of midnight January 1, 2024
    source.family.apply_hints(incremental=dlt.sources.incremental("updated"),initial_value=pendulum.DateTime(2024, 1, 1, 0, 0, 0))
    #running the pipeline
    info = pipeline.run(source, write_disposition="merge")
    print(info)
    ```  

   :::info

>    * For merge write disposition, the source table needs a primary key, which `dlt` automatically sets up. 

Reviewer: ```suggestion
   * When using "merge" write disposition, the source table needs a primary key, which `dlt` automatically sets up. 
```

   * `apply_hints` is a powerful method that enables schema modifications after resource creation, like adjusting write disposition and primary keys. You can choose from various tables and use `apply_hints` multiple times to create pipelines with merged, appended, or replaced resources.
   :::

## Parallelized extraction

You can extract each table in a separate thread (no multiprocessing at this point). This will decrease loading time if your queries take time to execute or your network latency/speed is low. To enable this, declare your sources/resources as follows:
```py
database = sql_database().parallelize()
table = sql_table().parallelize()
```

> ## Column reflection

Reviewer: I would add a sentence here quickly explaining reflection
